README.md

```diff
@@ -218,6 +221,70 @@ if (entity != null) {
 }
 ```
 
+Google Cloud DNS (Alpha)
+----------------------
+- [API Documentation][dns-api]
+- [Official Documentation][cloud-dns-docs]
+
+*Follow the [activation instructions][cloud-dns-activation] to use the Google Cloud DNS API with your project.*
+
+#### Preview
+
+Here are two code snippets showing simple usage examples from within Compute/App Engine. Note that you must [supply credentials](#authentication) and a project ID if running this snippet elsewhere.
+
+The first snippet shows how to create a zone resource. Complete source code can be found on
+[CreateZone.java](./gcloud-java-examples/src/main/java/com/google/gcloud/examples/dns/snippets/CreateZone.java).
+
+```java
+import com.google.gcloud.dns.Dns;
+import com.google.gcloud.dns.DnsOptions;
+import com.google.gcloud.dns.Zone;
+import com.google.gcloud.dns.ZoneInfo;
+
+Dns dns = DnsOptions.defaultInstance().service();
+String zoneName = "my-unique-zone";
+String domainName = "someexampledomain.com.";
+String description = "This is a gcloud-java-dns sample zone.";
+ZoneInfo zoneInfo = ZoneInfo.of(zoneName, domainName, description);
+Zone zone = dns.create(zoneInfo);
+```
+
+The second snippet shows how to create records inside a zone. The complete code can be found on [CreateOrUpdateDnsRecords.java](./gcloud-java-examples/src/main/java/com/google/gcloud/examples/dns/snippets/CreateOrUpdateDnsRecords.java).
+
+```java
+import com.google.gcloud.dns.ChangeRequest;
+import com.google.gcloud.dns.Dns;
+import com.google.gcloud.dns.DnsOptions;
+import com.google.gcloud.dns.DnsRecord;
+import com.google.gcloud.dns.Zone;
+
+import java.util.Iterator;
+import java.util.concurrent.TimeUnit;
+
+Dns dns = DnsOptions.defaultInstance().service();
+String zoneName = "my-unique-zone";
+Zone zone = dns.getZone(zoneName);
+String ip = "12.13.14.15";
+DnsRecord toCreate = DnsRecord.builder("www.someexampledomain.com.", DnsRecord.Type.A)
+    .ttl(24, TimeUnit.HOURS)
+    .addRecord(ip)
+    .build();
+ChangeRequest changeRequest = ChangeRequest.builder().add(toCreate).build();
```

Comment:
Since we changed the snippet to include the loop, this needs to become:

`ChangeRequest.Builder changeBuilder = ChangeRequest.builder().add(toCreate);`